client/src/index.js

```diff
@@ -249,26 +210,28 @@ function createFS(options) {
 
     sync.state = sync.SYNC_DISCONNECTED;
     sync.emit('disconnected');
-    log.info('Disconnected');
+    log.info('Disconnected from MakeDrive server');
   };
 
   // Request that a sync begin.
+  // sync.request does not take any parameters
+  // as the path to sync is determined internally
   sync.request = function() {
-    // sync.request does not take any parameters
-    // as the path to sync is determined internally
-    // requestSync on the other hand optionally takes
-    // a path to sync which can be specified for
-    // internal use
-    requestSync();
+    if(sync.state === sync.SYNC_CONNECTING || sync.state === sync.SYNC_DISCONNECTED) {
+      sync.emit('error', new Error('MakeDrive error: MakeDrive cannot sync as it is either disconnected or trying to connect'));
```

Comment:
I actually think that something more than a log should be done here.
What I was thinking was that in a separate bug, we could have error constants that each indicate something. In that case, the devs using our library can handle each error appropriately